src/cogent3/app/sqlite_data_store.py

```diff
@@ -295,6 +277,64 @@ def drop_not_completed(self) -> None:
         self.db.execute(f"DELETE FROM {_RESULT_TABLE} WHERE is_completed=0")
         self._not_completed = []
 
+    def lock(self):
+        """if writable, and not locked, locks the database to this pid"""
+        # if mode=w and the data store exists AND has a lock_pid
+        # value already, then we should fail. The user might
+        # inadvertently overwrite something otherwise.
+        # BUT if mode=a, as the user expects to be modifying the data
+        # store then we have to update the value
+        if self.mode is not READONLY:
+            result = (
+                None
+                if self.mode is APPEND
+                else self._db.execute("SELECT state_id,lock_pid FROM state")
+            ).fetchall()
+            unlocked = False
+            if result and len(result):
+                unlocked = True
+            locked = result[0]["lock_pid"] if result else None
+            if locked and self.mode is OVERWRITE:
+                raise IOError(
+                    "You are trying to OVERWRITE a database which is locked. Use APPEND mode or unlock"
+                )
+            elif locked or unlocked:
+                # we will update an existing
+                state_id = result[0]["state_id"]
+                cmnd = "UPDATE state SET lock_pid=? WHERE state_id=?"
+                vals = [os.getpid(), state_id]
+            else:
+                cmnd = "INSERT INTO state(lock_pid) VALUES (?)"
+                vals = [os.getpid()]
+
+            self._db.execute(cmnd, tuple(vals))
+
+    @property
+    def is_locked(self):
+        """returns lock pid or None if unlocked or pid matches self"""
```

Comment:
The expression returns a bool, so the docstring should be changed and a return type hint added